packages/modular-scripts/src/config/jest/index.js

```diff
@@ -0,0 +1,171 @@
+// We run jest by ourselves instead of using CRA's test runner because it assumes
+// that we're running from the context of an app, wherewas we're running the context
+// of a monorepo. Owning the runner then gives us the opportunity to generate
+// coverage reports correctly across workspaces, etc.
+'use strict';
+
+const path = require('path');
+const glob = require('glob');
+const chalk = require('chalk');
+const fs = require('fs-extra');
+
+const moduleFileExtensions = [
+  'web.mjs',
+  'mjs',
+  'web.js',
+  'js',
+  'web.ts',
+  'ts',
+  'web.tsx',
+  'tsx',
+  'json',
+  'web.jsx',
+  'jsx',
+];
+
+if (!process.env.MODULAR_ROOT) {
+  throw new Error(
+    // this should never be visible to a user, only us when we're developing
+    'MODULAR_ROOT not found in environment, did you forget to pass it when calling cracoBin in cli.ts?',
+  );
+}
+const modularRoot = process.env.MODULAR_ROOT;
+const absolutePackagesPath = path.resolve(modularRoot, 'packages');
+const absoluteModularGlobalConfigsPath = path.resolve(modularRoot, 'modular');
+
+const config = {
+  rootDir: absolutePackagesPath,
+  roots: ['<rootDir>'],
+  testMatch: ['<rootDir>/**/src/**/*.{spec,test}.{js,ts,tsx}'],
+  coverageDirectory: path.resolve(modularRoot, 'coverage'),
+  collectCoverageFrom: ['<rootDir>/**/src/**/*.{js,ts,tsx}', '!**/*.d.ts'],
+  setupFiles: [require.resolve('react-app-polyfill/jsdom')]
+    .concat([path.join(__dirname, './jest-setupEnvironment.js')])
+    .concat(
+      glob.sync(
+        `${absoluteModularGlobalConfigsPath}/setupEnvironment.{js,ts,tsx}`,
+        {
+          cwd: process.cwd(),
+        },
+      ),
+    ),
+  setupFilesAfterEnv: glob.sync(
+    `${absoluteModularGlobalConfigsPath}/setupTests.{js,ts,tsx}`,
+    {
+      cwd: process.cwd(),
+    },
+  ),
+
+  testEnvironment: 'jsdom',
+  testRunner: require.resolve('jest-circus/runner'),
```

Comment:
We need this to be configurable please since we still have a dependency on ```jest-jasmine2```! Or will it pick up overrides from ```package.json```? Something to check for internally..